### PR TITLE
Currency converter branch

### PR DESCRIPTION
**Pull request by hebaali9**

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,45 +1,29 @@
+@import url("https://fonts.googleapis.com/css2?family=Dancing+Script:wght@500&display=swap");
+
+/* * {
+  box-sizing: border-box;
+  margin: 0;
+  padding: 0;
+} */
```
Comment (SirajKakeh) [inline src/App.css:7]: let's keep the `box-sizing: border-box;` property

File: src/App.css [hunk 2]
```diff
@@ -1,45 +1,29 @@
...
-  height: 6em;
-  padding: 1.5em;
-  will-change: filter;
-}
-.logo:hover {
-  filter: drop-shadow(0 0 2em #646cffaa);
+/* header {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
```
Comment (SirajKakeh) [inline src/App.css:14]: let's remove unused comments

File: src/App.tsx [hunk 3]
```diff
@@ -10,21 +10,30 @@ import SignUp from "./siginUp";
 import LogIn from "./logIn";
 import ProtectedRoutes from "./protectedRoutes";
 import CurrencyPage from "./currencyPage";
+import { useSelector } from "react-redux";
 import { store } from "./store";
+type RootState = {
+  token: string;
+};
```
Comment (SirajKakeh) [inline src/App.tsx:17]: this seems like a global type related to the store, let's define in the store and export it from there. It seems like the wrong place to define it here

File: src/App.tsx [hunk 4]
```diff
@@ -10,21 +10,30 @@ import SignUp from "./siginUp";
...
+          <li>signup</li>
+        </Link>
         <br />
-        <Link to="/login">logIn</Link>
+        <Link to="/login">
+          <li>logIn</li>
+        </Link>
         <br />
-        <Link to="/currenctPage">CurrencyPage</Link>
+        <Link to="/currenctPage">
```
Comment (SirajKakeh) [inline src/App.tsx:34]: the route should be user-friendly. Let's name it something more user friendly

File: src/currencyPage.tsx [hunk 5]
```diff
@@ -1,28 +1,77 @@
...
-          <option value="hi"></option>
-        </select>
+      <br></br>
+      <form>
+        <Input
+          className="antd-input"
+          type="number"
+          value={fromAmount.toString()}
+          onChange={handleChange}
+          //onChange={(e) => setFromAmount(Number(e.target.value))}
```
Comment (SirajKakeh) [inline src/currencyPage.tsx:37]: unused comment

File: src/currencyPage.tsx [hunk 6]
```diff
@@ -1,28 +1,77 @@
...
+        ></Input>
+
+        <Select
+          value={fromCurrency}
+          // @ts-ignore
```
Comment (SirajKakeh) [inline src/currencyPage.tsx:42]: no need to ignore the error here, let's fix it instead

File: src/main.tsx [hunk 7]
```diff
@@ -1,10 +1,12 @@
 import React from "react";
 import ReactDOM from "react-dom/client";
+import { Provider } from "react-redux";
 import App from "./App";
 import "./index.css";
+import { store } from "./store";
 
 ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
-  <React.StrictMode>
+  <Provider store={store}>
```
Comment (SirajKakeh) [inline src/main.tsx:9]: let's keep React.StrictMode and nest `Provider` inside it

File: src/protectedRoutes.tsx [hunk 8]
```diff
@@ -1,9 +1,12 @@
...
-import { store } from "./store";
+import { Navigate } from "react-router-dom";
+import { useSelector } from "react-redux";
 
 function ProtectedRoutes({ children }: { children: JSX.Element }) {
-  const token = store.getState().token;
-  const isLogedIn = token;
+  type RootState = {
+    token: string;
+  };
```
Comment (SirajKakeh) [inline src/protectedRoutes.tsx:7]: same as above, let's move `RootState` definition to store files

File: src/protectedRoutes.tsx [hunk 9]
```diff
@@ -1,9 +1,12 @@
...
+  const token = useSelector((state: RootState) => state.token);
```
Comment (SirajKakeh) [inline src/protectedRoutes.tsx:8]: this selector is used in many places, let's define `useToken` function that does the same thing and use it everywhere we want to access the token

**Review by SirajKakeh — CHANGES_REQUESTED** [review]

File: src/App.css (hunk 2 above)
Comment (hebaali9) [reply src/App.css:14]: it is not unused i was trying to make a navbar

File: src/App.css (hunk 1 above)
Comment (hebaali9) [reply src/App.css:7]: Ma fhemtak please wade7

**Comment by hebaali9** [comment]
اعمل ميرج؟؟؟

**Pull request merged**